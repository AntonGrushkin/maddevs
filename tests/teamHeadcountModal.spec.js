import {
  mount
} from '@vue/test-utils';
import teamHeadcountModal from '@/components/Modals/team-headcount-modal';

describe('Team headcount modal', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(teamHeadcountModal, {
      stubs: ['ValidationProvider', 'ValidationObserver', 'modal', 'perfect-scrollbar']
    });
  });

  // ------ IMPORTANT ----- //
  test('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy();
  });
  
  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
  // --------------------- //

  test('sets the correct default data', () => {
    expect(typeof teamHeadcountModal.data).toBe('function');
    const defaultData = teamHeadcountModal.data();
    expect(
      defaultData.agreeWithPrivacyPolicy &&
      defaultData.agreeToGetMadDevsDiscountOffers
    ).toEqual(false);
    expect(defaultData.inputId).toEqual('team-headcount');
  });

  test('has a functions', () => {
    expect(
      typeof teamHeadcountModal.methods.getPrivacyCheckboxState && 
      typeof teamHeadcountModal.methods.getDiscountOffersCheckboxState
    ).toBe('function');
  });

  test('call functions with params and change variables state', () => {
    wrapper.vm.getPrivacyCheckboxState(true);
    wrapper.vm.getDiscountOffersCheckboxState(true);

    expect(
      wrapper.vm.$data.agreeWithPrivacyPolicy &&
      wrapper.vm.$data.agreeToGetMadDevsDiscountOffers
    ).toEqual(true);
  });
});
