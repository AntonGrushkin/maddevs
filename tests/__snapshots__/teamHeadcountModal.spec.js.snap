// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Team headcount modal renders correctly 1`] = `
<modal-stub
  classes="modal_container"
  height="auto"
  name="team-headcount-modal"
>
  <img
    alt="Close modal"
    class="close-modal"
    src="@/assets/img/common/close-icon.svg"
  />
   
  <perfect-scrollbar-stub
    class="modal_scrollbar"
    options="[object Object]"
  >
    <validationobserver-stub />
  </perfect-scrollbar-stub>
</modal-stub>
`;
